torch/ao/quantization/pt2e/utils.py

```diff
@@ -11,12 +11,83 @@
 from torch.utils._pytree import LeafSpec
 from torch._export import capture_pre_autograd_graph
 
+from torch.ao.quantization.quantizer import QuantizationAnnotation
+from torch.ao.quantization.quantizer.utils import _is_sym_size_node
+
 __all__ = [
     "fold_bn_weights_into_conv_node",
     "get_aten_graph_module",
     "remove_tensor_overload_for_qdq_ops",
+    "filter_sym_size_users",
+    "find_q_dq_node_for_user",
+    "is_valid_annotation",
+]
+
+_QUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.quantize_per_tensor.default,
+    torch.ops.quantized_decomposed.quantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.quantize_per_channel.default,
 ]
 
+
+_DEQUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.dequantize_per_tensor.default,
+    torch.ops.quantized_decomposed.dequantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.dequantize_per_channel.default,
+]
+
+
+def _is_connected(next_node: torch.fx.Node, target: torch.fx.Node) -> bool:
+    if target.op == "output":
+        return False
+    if next_node == target:
+        return True
+    for n in next_node.users.keys():
+        if _is_connected(n, target):
+            return True
+    return False
+
+
+def find_q_dq_node_for_user(
```

Comment:
I feel this function is a bit too complicated, is there a way to simplify this?

Reply:
it is just finding q, dq from [producer -> .... -> q -> dq - > user]. I can add doc string if you want

Reply:
yeah a docstring would be helpful